app/agents/RecruitmentManagerAgent.py

```diff
@@ -1,27 +1,249 @@
+        self.recruitment_instruction_module = RecruitmentInstructionModule(
+            self.agent_config.dbname, self.logger
+        )
+        self.if_created = False
+        self.recruitment: Recruitment = None
+        self.recruitment_instruction: RecruitmentInstruction = None
 
-        # TODO behaviours
+        # behaviours
+        self.check_recruitments_behav: CheckRecruitments = None
+        self.prepare_recruitment_behav: PrepareRecruitment = None
+        self.stage_communication_behav: StageCommunication = None
 
     async def setup(self):
         await super().setup()
-        self.recruitment = self.recruitmentModule.get_stages_info(self.job_offer_id)
-        if self.recruitment is None:
-            await self.stop()
 
-        # TODO
-        # self.add_behaviour()
+        self.check_recruitments_behav = CheckRecruitments()
+        self.stage_communication_behav = StageCommunication()
+
+        self.add_behaviour(self.check_recruitments_behav)
+        self.add_behaviour(self.stage_communication_behav)
+
+
+class CheckRecruitments(spade.behaviour.OneShotBehaviour):
+    """Checks whether recruitments with given job_offer_id, candidate_id and required stages are present in db"""
+
+    agent: RecruitmentManagerAgent
+
+    async def run(self):
+        self.agent.logger.info("CheckRecruitments behaviour run.")
+
+        await self.check_recruitments()
+
+        self.agent.prepare_recruitment_behav = PrepareRecruitment()
+        self.agent.add_behaviour(self.agent.prepare_recruitment_behav)
+
+    async def check_recruitments(self):
+        recruitments = self.agent.recruitment_module.get_by_job_and_candidate(
+            self.agent.job_offer_id, self.agent.candidate_id
+        )
+
+        if len(recruitments) == 0:
+            self.agent.if_created = False
+            self.agent.logger.info(
+                f"No recruitments found. Recruitment with job_offer_id: {self.agent.job_offer_id} and candidate_id: {self.agent.candidate_id} to be created."
+            )
+        else:
+            self.agent.if_created = True
+            self.agent.recruitment = recruitments[0]
+
+            self.agent.logger.info(
+                f"Found recruitment with id: {recruitments[0]._id}. No recruitment objects will be created."
+            )
+
+
+class PrepareRecruitment(spade.behaviour.OneShotBehaviour):
+    """Creates recruitment object (if not present) and initiates RmentStageAgents (one per one stage)"""
+
+    agent: RecruitmentManagerAgent
+
+    async def run(self):
+        self.agent.logger.info("PrepareRecruitment behaviour run.")
+
+        await self.get_recruitment_instruction()
+        await self.create_recruitment()
+        await self.create_stage_agents()
+
+    async def get_recruitment_instruction(self):
+        self.agent.recruitment_instruction = (
+            self.agent.recruitment_instruction_module.get_by_job_offer_id(
+                self.agent.job_offer_id
+            )
+        )
```

Comment:
1. nie widzę nigdzie walidacji czy ta instrukcja istnieje, jeśli nie to agent powinien się zastopować z błędem
2. myślę że to powinno być w poprzednim behavie, bardziej na początku, a nawet może w setupie całego agenta bo to jest dość podstawowa sprawa i bez niej nie ma sensu nic robić

Reply:
1. nie robilem sprawdzen instrukcji bo ona de facto sie tworzy przy inicie bazy (skrypt init_aasd_db.js), ale jasne, dorzuce walidacje

Reply:
2. rozbilem logicznie sprawdzenie czy obiekty sa juz w bazie i ewentualne tworzenie nowych obiektow plus rsm agentow na dwa zachowania (wtedy mi sie to wydawalo logiczniejsze by nie pakowac wszystkiego do jednego)